### DISCUSSION TITLE
How to rename terminal tab programmatically?

### DISCUSSION THREAD

**Discussion by mmx85** [eclipse-cdt/cdt#492] [Q&A] [answered]
I have the code that setups terminal and opens it with custom tab name after some user actions. It looks like this:

HashMap<String, Object> params = new HashMap<String, Object>();
params.put(ITerminalsConnectorConstants.PROP_TITLE, "some name"));
params.put(ITerminalsConnectorConstants.PROP_DELEGATE_ID, delegateId);
params.put(ITerminalsConnectorConstants.PROP_PROCESS_PATH, bashShellExecutablePath);
params.put(ITerminalsConnectorConstants.PROP_PROCESS_WORKING_DIR, workingDir);
...
TerminalServiceFactory.getService().openConsole(params, null);

It worked well for me with CDT 10.0.

After updating to CDT 10.2 and higher the code above sets the correct name but VT100Emulator.processNewText() overrides it reading a startup sequence that Bash outputs with a command to change terminal title to current working directory value. When VT100Emulator reads this command it overrides terminal tab title with current working directory value calling: processNewText()->processAnsiOsCommand()->terminal.setTerminalTitle(ansiOsCommand.substring(2)) in VT100Emulator class. 
This is cased by the following changes: https://git.eclipse.org/c/cdt/org.eclipse.cdt.git/commit/?id=fe003208544902646096db658f63d2038a80d40a
Probably now there is a different approach how to set custom terminal tab name in a way that it is not overridden but it is not obvious or this use case was not considered when renaming feature was implemented. Any ideas how to set custom terminal tab name in a way that it is not overridden?

**Comment by jonahgraham** [comment, marked as answer]
Yes - renaming terminal being something user can do from the UI or from escape sequences was a new feature added in CDT 10.2 ([N&N entry](https://github.com/eclipse-cdt/cdt/blob/main/NewAndNoteworthy/CDT-10.2.md#rename-terminal-tab))

> Any ideas how to set custom terminal tab name in a way that it is not overridden?

I don't think so, a PR adding this feature is very welcome.

**Reply by mmx85** [reply]
After examining the source code I think it make sense to add a ITerminalsConnectorConstants.PROP_TITLE_UPDATE constant with options:
1. PROP_TITLE_UPDATE_API (to enable title change from menu and with value set to ITerminalsConnectorConstants.PROP_TITLE parameter but disable ability to change title with ANSI command "\e]0;...\u0007") and 
2. PROP_TITLE_UPDATE_MIXED (keep current behavior).

**Comment by ethanyhou** [comment]
If anyone has similar issue, you can setup via ```properties.put(ITerminalsConnectorConstants.PROP_TITLE_DISABLE_ANSI_TITLE, true);```. It works for me.
